src/templates/base/2019/base.html

```diff
@@ -99,11 +99,134 @@
+          </option>
+        {% endif %}
+        {% endif %}
+      {% endif %}
+    {% endfor %}
+    {% endfor %}
+
+    <option value="{{ url_for('methodology', year=year, lang=lang) }}">
+      {{ self.methodology_title() }}
+    </option>
+    <option value="{{ url_for('contributors', year=year, lang=lang) }}">
+      {{ self.contributors_title() }}
+    </option>
+    <option value="/static/pdfs/web_almanac_{{ year }}_{{ lang }}.pdf">
+      {{ self.ebook_title() }}
+    </option>
+  </select>
+</div>
+{% endmacro %}
+
+{% macro styled_table_of_contents_dropdown(switcher_name) %}
+{% set current_title = metadata.get('title') if metadata else None %}
+<div class="nav-dropdown {{switcher_name}} table-of-contents">
+  <button class="nav-dropdown-btn enable-script" aria-expanded="false" aria-label="{{ self.table_of_contents_title() }}" >
+    <a class="js-enable-script-remove-tag" href="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.table_of_contents_title() }}</a>
+  </button>
+  <ul class="nav-dropdown-list hidden {{switcher_name}}-list">
+    
+    <li class="nav-dropdown-list__chapter foreword">
+      <a href="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.foreword_title() }}</a>
+    </li>
+
+    {% for part_config in config.outline %}
+    <li class="nav-dropdown-list__part">
+      {{ self.part() }} {{ localizedPartTitles[part_config.part] if localizedPartTitles[part_config.part]|length else chapter_config.title }}
+    </li>
+    {% for chapter_config in part_config.chapters %}
+    {% set title = localizedChapterTitles[chapter_config.slug] if localizedChapterTitles[chapter_config.slug]|length else chapter_config.title %}
+    <li class="nav-dropdown-list__chapter {% if current_title == title %}active{% endif %}">
+      {% if chapter_config.todo %}
+      <span class="nav-dropdown-list__todo">{{ self.chapter() }} {{ chapter_config.chapter }}: {{ title }}</span>
+      {% else %}
+        {% if chapter_lang_exists(lang, year, chapter_config.slug) %}
+        <a href="{{ url_for('chapter', year=year, lang=lang, chapter=chapter_config.slug) }}">
+          {{ self.chapter() }} {{ chapter_config.chapter }}: {{ title }}
+        </a>
+        {% else %}
+        <a href="{{ url_for('chapter', year=year, lang='en', chapter=chapter_config.slug) }}">
+          {{ self.chapter() }} {{ chapter_config.chapter }}: {{ title }} <span class="not-translated">({{ self.translation_not_available() }})</span>
+        </a>
+        {% endif %}
+      {% endif %}
+    </li>
+    {% endfor %}
+    {% endfor %}
+
+    <li class="nav-dropdown-list__part">
+      {{ self.appendices() }}
+    </li>
+    <li class="nav-dropdown-list__chapter">
+      <a href="{{ url_for('methodology', year=year, lang=lang) }}">{{ self.methodology_title() }}</a>
+    </li>
+    <li class="nav-dropdown-list__chapter">
+      <a href="{{ url_for('contributors', year=year, lang=lang) }}">{{ self.contributors_title() }}</a>
+    </li>
```

Comment:
Grey out if current page:

```suggestion
    <li class="nav-dropdown-list__chapter">
      {% if request.path.endswith("methodology") %}
      <span class="nav-dropdown-list__current">{{ self.methodology_title() }}</span>
      {% else %}
      <a href="{{ url_for('methodology', year=year, lang=lang) }}">{{ self.methodology_title() }}</a>
      {% endif %}
    </li>
    <li class="nav-dropdown-list__chapter">
      {% if request.path.endswith("contributors") %}
      <span class="nav-dropdown-list__current">{{ self.contributors_title() }}</span>
      {% else %}
      <a href="{{ url_for('contributors', year=year, lang=lang) }}">{{ self.contributors_title() }}</a>
      {% endif %}
    </li>
```

Reply:
i fixed.
I fixed part deep links.
https://github.com/HTTPArchive/almanac.httparchive.org/pull/1393#discussion_r516162235